amITalkingALot.js

```diff
@@ -114,36 +95,56 @@
 
     indicators.forEach((node, i) => {
       const participant = participants[i];
+
       if (!participant) {
-        console.error("No matching participant for indicator", {
+        debug("No matching participant for indicator", {
           index: i,
           indicators,
           node,
           participants,
         });
         return;
       }
+
       const name = participant.textContent;
-      if (!tally[name]) {
+
+      if (name.includes("(Present")) {
+        debug("Unexpected participant: " + name, {
+          indicators,
+          participants,
+        });
+      }
+
+      if (!tally.hasOwnProperty(name)) {
+        debug(`Logging new participant: ${name}`);
         tally[name] = 0;
       }
-      if (alphaStringifyClassList(node.classList) !== quietClassString) {
+
+      // If the second bar in the indicator is taller than 4px, then we know
+      // that the participant is talking.
+      // If they are muted, the value will be 0.
+      // If they are unmuted and not talking, the value will be 4.
+      if (node.children[1].getBoundingClientRect().height > 4) {
```

Comment:
This is the real change here. Instead of trying to compare lists of compiled class names, here we are measuring a UI element to determine if a participant is speaking.

This will break if Google Meet changes their UI, but so would the previous solution. For the time being, this is more accurate than the previous solution.

Reply:
Appreciate the resourcefulness! Thanks for picking this up again, Sean! I've missed the feedback ;-)